core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/dr/BackupRestoreSaltConfigGenerator.java

```diff
@@ -43,7 +56,17 @@ public SaltConfig createSaltConfig(String location, String backupId, String rang
         disasterRecoveryValues.put(OBJECT_STORAGE_URL_KEY, fullLocation);
         disasterRecoveryValues.put(RANGER_ADMIN_GROUP_KEY, rangerAdminGroup);
         disasterRecoveryValues.put(CLOSE_CONNECTIONS, String.valueOf(closeConnections));
-
+        if (!skipDatabaseNames.isEmpty()) {
+            ArrayList<String> names = new ArrayList<>();
+            DEFAULT_BACKUP_DATABASE.forEach(db -> names.add(db.toString().toLowerCase()));
+            skipDatabaseNames.forEach(db -> {
+                if (!names.remove(db)) {
+                    LOGGER.warn("Tried to skip unknown database " + db);
+                }
+            });
+            names.removeAll(skipDatabaseNames);
```

Comment:
I'd prefer not mixing functional and procedural style, this could be rewritten e.g. like:

```
            List<String> names = DEFAULT_BACKUP_DATABASE.stream()
                    .map(DatabaseType::toString)
                    .map(String::toLowerCase)
                    .collect(Collectors.toList());

            
            skipDatabaseNames.stream()
                    .filter(Predicate.not(names::contains))
                    .forEach((this::log));


    private void log(String db) {
        LOGGER.warn("Tried to skip unknown database " + db);
    }
            
```

Reply:
Updated much better. Also that removeAll was redundant.